experiments/mrmc_experiment/run_mrmc_experiment.py

```diff
@@ -0,0 +1,455 @@
+            for.
+        dataset_name: The name of the dataset to use.
+        model_type: The type of model to use.
+        cluster_seed: The seed to use for clustering.
+        _unused_kwargs: An argument used to capture kwargs from run_config that
+            aren't used by this function.
+
+    Returns:
+        A list of recourse paths and a dataframe containing cluster info."""
+    # generate random seeds
+    poi_seed, adapter_seed = np.random.default_rng(run_seed).integers(
+        0, 10000, size=2
+    )
+
+    # initialize dataset, adapter, model, mrmc, and recourse iterator
+    dataset, dataset_info = _get_dataset(dataset_name)
+    adapter = _get_recourse_adapter(
+        dataset=dataset,
+        dataset_info=dataset_info,
+        random_seed=adapter_seed,
+        noise_ratio=noise_ratio,
+        rescale_ratio=rescale_ratio,
+    )
+    model = _get_model(model_type, dataset_name)
+    mrmc = _get_mrmc(
+        dataset=dataset,
+        adapter=adapter,
+        model=model,
+        num_paths=num_paths,
+        volcano_cutoff=volcano_cutoff,
+        volcano_degree=volcano_degree,
+        step_size=step_size,
+        confidence_threshold=confidence_cutoff,
+        random_seed=cluster_seed,
+    )
+    iterator = _get_recourse_iterator(adapter, mrmc, confidence_cutoff, model)
+
+    # get the POI
+    poi = utils.random_poi(
+        dataset,
+        label_column=dataset_info.label_column,
+        label_value=adapter.negative_label,
+        random_seed=poi_seed,
+    )
+
+    # generate the paths
+    paths = iterator.iterate_k_recourse_paths(
+        poi=poi, max_iterations=max_iterations
+    )
+
+    # retrieve the clusters
+    clusters = mrmc.clusters.cluster_centers
+    cluster_df = pd.DataFrame(
+        data=clusters,
+        columns=adapter.embedded_column_names(),
+    )
+
+    return paths, cluster_df
+
+
+def format_results(
+    mrmc_paths: Sequence[pd.DataFrame],
+    clusters_df: pd.DataFrame,
+    run_config: Mapping[str, Any],
+) -> Mapping[str, pd.DataFrame]:
+    """Formats the results as DataFrames ready for analysis.
+
+    It adds the path_id and step_id keys to the mrmc_paths dataframe. It also
+    adds keys from the experiment_utils.format_results() function.
+
+    Args:
+        mrmc_paths: The list of path dataframes output by recourse iteration.
+        clusters_df: A dataframe containing cluster information.
+        run_config: The run config used to generate these results.
+
+    Returns:
+        A mapping from dataframe name to formatted dataframe."""
+    clusters_df["path_id"] = np.arange(len(clusters_df))
+    for i, path in enumerate(mrmc_paths):
+        path["step_id"] = np.arange(len(path))
+        path["path_id"] = i
+    mrmc_paths_df = pd.concat(mrmc_paths).reset_index(drop=True)
+    index_df, clusters_df, mrmc_paths_df = experiment_utils.format_results(
+        run_config, clusters_df, mrmc_paths_df
+    )
+    return {
```

Comment:
This is (mostly) how the output works. The differences are:

* The experiment_id is called batch_id. This is terrible and will be changed later with [issue 41](https://github.com/jakeval/MRMC/issues/41).
* cluster_df saves the centroids for each cluster _per-run_. This is because the configuration setup allows users to provide different cluster seeds for every run if they desire to. It's cumbersome but works well enough.